### PR TITLE
fix(sentry): Add sampler to fuction to block /health transactions

### LINKED ISSUE #228: Don't send /health healthcheck transactions to Sentry
We need to set up a sampling function as per https://docs.sentry.io/platforms/python/guides/django/configuration/sampling/#setting-a-sampling-function such that calls to the /health endpoint are never sent to Sentry as they are consuming too much of our quota.

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

File: api/integrations/sentry/apps.py [hunk 1]
```diff
@@ -12,9 +12,17 @@ def ready(self):
...
-                traces_sample_rate=settings.SENTRY_TRACE_SAMPLE_RATE,
                 environment=settings.ENV,
+                traces_sampler=traces_sampler,
                 # If you wish to associate users to errors (assuming you are using
                 # django.contrib.auth) you may enable sending PII data.
                 send_default_pii=True,
             )
+
+
+def traces_sampler(ctx):
```
Comment (matthewelwell) [inline api/integrations/sentry/apps.py]: I'm not a massive fan of this being defined in the `apps` module. Could we move it somewhere more appropriate? `samplers` or something?

File: api/integrations/sentry/samplers.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.conf import settings
+
+
+def block_health_sampler(ctx):
```
Comment (matthewelwell) [inline api/integrations/sentry/samplers.py:4]: Could we add a unit test for this now?

**Review by matthewelwell — CHANGES_REQUESTED** [review]

**Comment by dabeeeenster** [comment]
Actually sorry @gagantrivedi but this needs to move out of the `integrations/sentry` area altogether. There are 2 ways Flagsmith uses sentry:

1. As an APM for the API itself. This PR is related to that. 
2. For customers of the platform to send flag change events into *their* sentry installation. This is what `integrations/sentry` is for. 

I think this code should live in `app` - what do you think @matthewelwell ?

Having said that - I've tested this and its working for me - stripping out health endpoint checks correctly. It just needs moving out of integrations.

**Comment by matthewelwell** [comment]
I don't really see any reason why it needs to move out of the `sentry` package tbh? It's still code related to Sentry, it's just that there are multiple ways in which we use Sentry.

**Comment by dabeeeenster** [comment]
OK lets leave it here - but add that test pls @gagantrivedi

**Comment by gagantrivedi** [comment]
> OK lets leave it here - but add that test pls @gagantrivedi

Sure, but is this working for you locally? i.e: can you see transaction generated locally on the sentry dashboard?

**Comment by dabeeeenster** [comment]
Yes - transactions other than /health show up

File: api/integrations/sentry/tests/test_sampler.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
+from django.conf import settings
+
+from integrations.sentry.samplers import block_health_sampler
+
+
+def test_block_health_sampler_empty_context_returns_defaut_sample_rate():
+    # Given
+    settings.SENTRY_TRACE_SAMPLE_RATE = 10
```
Comment (matthewelwell) [inline api/integrations/sentry/tests/test_sampler.py]: It's better to use the `@override_settings()` decorator to do this so your tests become something like the following: 

```python
sample_rate = 10

@override_settings(SENTRY_TRACE_SAMPLE_RATE=sample_rate)
def test_block_health_sampler_empty_context_returns_defaut_sample_rate():
    assert block_health_sampler({}) == sample_rate
```

See documentation links here: 
 - https://docs.djangoproject.com/en/3.2/topics/settings/#altering-settings-at-runtime
 - https://docs.djangoproject.com/en/3.2/topics/testing/tools/#django.test.override_settings

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: api/integrations/sentry/tests/test_sampler.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+SAMPLE_RATE = 10
+
+
+@override_settings(SENTRY_TRACE_SAMPLE_RATE=SAMPLE_RATE)
+def test_block_health_sampler_empty_context_returns_defaut_sample_rate():
+    # When
+    sample_rate = block_health_sampler({})
+
+    # Then
+    assert sample_rate == 10
```
Comment (matthewelwell) [inline api/integrations/sentry/tests/test_sampler.py:14]:
```suggestion
    assert sample_rate == SAMPLE_RATE
```

File: api/integrations/sentry/tests/test_sampler.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+@override_settings(SENTRY_TRACE_SAMPLE_RATE=SAMPLE_RATE)
+def test_block_health_sampler_returns_default_for_non_heath_check_transaction():
+    # Given
+    ctx = {"wsgi_env": {"PATH_INFO": "/home"}}
+
+    # When
+    sample_rate = block_health_sampler(ctx)
+
+    # Then
+    assert sample_rate == 10
```
Comment (matthewelwell) [inline api/integrations/sentry/tests/test_sampler.py:38]:
```suggestion
    assert sample_rate == SAMPLE_RATE
```

**Review by matthewelwell — APPROVED** [review]
Some minor changes so that we're reusing the variable we defined as much as possible.

**Pull request merged**